lib/tesla/adapter/mint.ex

```diff
@@ -310,18 +310,21 @@ if Code.ensure_loaded?(Mint.HTTP) do
       end
     end
 
-    defp receive_message(conn, %{mode: :active} = opts) do
-      receive do
-        message ->
-          HTTP.stream(conn, message)
-      after
-        opts[:timeout] -> {:error, :timeout}
+    defp receive_message(conn, opts) do
+      case opts[:mode] do
+        :active ->
+          receive do
+            message ->
+              HTTP.stream(conn, message)
+          after
+            opts[:timeout] -> {:error, :timeout}
+          end
+
+        :passive ->
+          HTTP.recv(conn, 0, opts[:timeout])
```

Comment:
This code looks identical, why was the change required? 🤔

Reply:
Because `opts` is passed as a keyword list without being parsed into a map into `read_chunk/3`.

Reply:
i've added [fix](https://github.com/teamon/tesla/pull/395) for this inconsistency

Reply:
Good work, I will remove this commit then